proto/src/determined/trial/v1/trial.proto

```diff
@@ -9,47 +9,67 @@ import "determined/checkpoint/v1/checkpoint.proto";
 import "google/protobuf/timestamp.proto";
 import "protoc-gen-swagger/options/annotations.proto";
 
-
-// TrialValidation represents a validation entry presenting the searcher_metric.
-message TrialValidation {
+// CheckpointWorkload is an artifact created by a trial during training.
+message CheckpointWorkload {
   option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
       json_schema: {
-      required: ["batch_number", "start_time", "state"]
+      required: ["start_time", "state"]
    }
   };
-  // The batch number where validation was evaluated.
-  int32 batch_number = 1;
-  // The time validation was started.
+  // UUID of the checkpoint.
+  string uuid = 1;
+  // The time the workload was started.
   google.protobuf.Timestamp start_time = 2;
-  // The time validaiton ended.
+  // The time the workload finished or was stopped.
   google.protobuf.Timestamp end_time = 3;
+  // The state of the checkpoint.
+  determined.checkpoint.v1.State state = 4;
+  // Dictionary of file paths to file sizes in bytes of all files in the checkpoint.
+  map<string, int64> resources = 5;
+}
+
+// MetricsWorkload is a workload generating metrics.
+message MetricsWorkload {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+      json_schema: {
+      required: ["start_time", "state"]
+   }
+  };
+  // The time the workload was started.
+  google.protobuf.Timestamp start_time = 1;
+  // The time the workload finished or was stopped.
+  google.protobuf.Timestamp end_time = 2;
   // The current validation state.
-  determined.experiment.v1.State state = 4;
-  // The value of the `searcher.metric`, indicated by the experiment config, for the validation.
-  float searcher_metric = 5;
+  determined.experiment.v1.State state = 3;
+  // Metrics.
+  map<string, float> metrics = 4; // FIXME string, string legacy metrics?
+  // Number of inputs processed.
+  int32 num_inputs = 5;
 }
 
-// CheckpointBase is an artifact created by a trial during training.
-message TrialCheckpoint {
+
+// Workload is a unit of work in Determined.
+message Workload {
```

Comment:
In hindsight, rereading this code, I'm sort of under the impression this should still be called step and that it should be described very clearly. If users use us enough to understand the internals over our system, this is fairly confusing because we're calling it a workload but it is exactly a step and it is comprised of workloads.

I think there are two ways forward: calling this step or putting step_id, num_batches and prior_batches_processed into each Workload above and then returning a list of repeated workloads (where they're actually workloads). Lmk what you think.

Reply:
thanks that's a good point.
calling it "step" would be simpler and closer to what we currently have. Going with separate workloads I think would help us 
1. do away with steps
2. give us better control over what workloads we are returning to the user. 

but also 
- means duplicate information for the current endpoint and use case
- would mean more changes for the current webui to adapt to.
- harder to associated workloads done in one _step_

Reply:
> means duplicate information for the current endpoint and use case

I'm not really concerned about the dupes if it's clearer.

> would mean more changes for the current webui to adapt to.

definitely should consider this.

> harder to associated workloads done in one step

I think we could remedy this by still including step id and calling it like, _workload_correlation_id or something weird

___

I think long term, calling it step again is probably a mistake, so we should probably go the workloads route.

Reply:
> harder to associated workloads done in one step
>> I think we could remedy this by still including step id and calling it like, _workload_correlation_id or something weird

I included num_batches and prior_batches_processed. that should give us enough information